tools/devops/provision-python-packages.csx

```diff
@@ -0,0 +1,8 @@
+using Xamarin.Provisioning;
+
+ProvisioningScript.ElevatedExec (
```

Comment:
If I understand correctly, we are installing this system wide. That would be a mistake. A better approach is to use a python virtualenv that inherits from the system based installed python modules.

For example:

```chsarp
pip install virtualenv
virtualenv "$VIRTUAL_ENV_PATH" --system-site-packages
source "$VIRTUAL_ENV_PATH/bin/activate"
pip install python-magic
```

The issue with using the activate in the virtual env, is that it will need to be done in the same exec process of the bash when we want to execute it. That is, we cannot do the virtual env activation in a step, then use python in a diff step. The correct possible approach is to do:

1. Setup a virtual env with dependencies.
2. Spawn a bash step that
    1. Activate the virtual env
    2. Executes python

If we are using bask steps on vsts, we wont need to call deactivate, although good citizens clean after them. It would be ideal that we have a way to provisionator to do the python installation, the virtualenv installation via pip (is not a default module in python2 AFAIK) and does call pip install on a freeze like the one we have. Then, that step should output a env variable with the virtual env. Ideally, we want the output variable in the provisionator step in case we want to setup more than one virtual env.

The issue I have with this approach, although it does work atm, is that it clutters the system and more importantly, the pipeline has side effects in the system. I do know that we already have side effects with brew.. and that is something we have to live with until we move to maybe using [chroot](https://linux.die.net/man/1/chroot) on Mac OS X + provisionator + brew (would be great to have [jails like in freebsd](https://www.freebsd.org/doc/handbook/jails.html), but Mac OS X is not really that cool).

Now that I think about it, provisionator with chroot + brew and vsts integration would be a really really amazing product. But then, I'm not PM ;)